server/include/map.h

```diff
@@ -59,11 +64,76 @@ extern "C" {
      * The map has a width, a height, and a 2D array of tiles.
      */
     typedef struct map_s {
-        int width;
-        int height;
-        tile_t **tiles;
+        size_t width;
+        size_t height;
+        array_t *tiles;
     } map_t;
 
+    /**
+     * @brief Init a new map.
+     *
+     * @param width The width of the map.
+     * @param height The height of the map.
+     * @return The new map.
+     */
+    map_t *init_map(size_t width, size_t height);
+
+    /**
+     * @brief deslete a map.
+     *
+     * @param map The map to free.
+     */
+    void destroy_map(map_t *map);
+
+    /**
+     * @brief Get a tile from the map.
+     *
+     * @param map The map.
+     * @param x The x coordinate of the tile.
+     * @param y The y coordinate of the tile.
+     * @return The tile.
+     */
+    tile_t *get_tile(map_t *map, size_t x, size_t y);
+
+    /**
+     * @brief Get a tile from the map.
+     *
+     * @param map The map.
+     * @param coordinates The coordinates of the tile.
+     * @return The tile.
+     */
+    tile_t *get_tile_by_coordinates(map_t *map, coordinates_t coordinates);
+
+    /**
+     * @brief Init a new tile.
+     *
+     * @param x The x coordinate of the tile.
+     * @param y The y coordinate of the tile.
+     * @return The new tile.
+     */
+    tile_t *init_tile(size_t x, size_t y);
```

Comment:
I don't get why you have a "coordinates_t" type, but you use it sometimes and sometimes not ?